frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx

```diff
@@ -25,29 +28,44 @@ interface SearchValuePickerProps {
 export function SearchValuePicker({
   fieldId,
   searchFieldId,
-  fieldValues: initialFieldValues,
   selectedValues,
-  placeholder,
+  columnDisplayName,
   shouldCreate,
   autoFocus,
   onChange,
 }: SearchValuePickerProps) {
   const [searchValue, setSearchValue] = useState("");
   const [searchQuery, setSearchQuery] = useState(searchValue);
+  const canSearch = searchQuery.length > 0;
 
-  const { data: fieldValues = initialFieldValues } = useSearchFieldValuesQuery(
+  const {
+    data: fieldValues = [],
+    error: searchError,
+    isFetching: isSearching,
+  } = useSearchFieldValuesQuery(
     {
       fieldId,
       searchFieldId,
       value: searchQuery,
       limit: SEARCH_LIMIT,
     },
     {
-      skip: !searchQuery,
+      skip: !canSearch,
     },
   );
 
-  const options = useMemo(() => getFieldOptions(fieldValues), [fieldValues]);
+  const searchOptions = canSearch ? getFieldOptions(fieldValues) : [];
+  const visibleOptions = getFilteredOptions(
+    searchOptions,
+    searchValue,
+    selectedValues,
+  );
```

Comment:
I'm not confident it's a good idea to have FE-side filtering on top of BE-side filtering.
I see where this is coming from but it makes for a bad UX in some of the cases.
I tried it without `getFilteredOptions` and I think it works way more smooth this way.
Please consult product/design team if you disagree.

There's an annoying flicker of dropdown content on subsequent searches (but not the first one).

https://github.com/user-attachments/assets/f24d417b-3bd8-4bac-84d4-1b131288a5bd

Reply:
I reported it separately: #45473